test/direct_uploads_controller_test.rb

```diff
@@ -12,6 +12,7 @@ class ActiveStorage::DirectUploadsControllerTest < ActionController::TestCase
       @blob = create_blob
       @routes = Routes
       @controller = ActiveStorage::DirectUploadsController.new
+      @config = SERVICE_CONFIGURATIONS
```

Comment:
IMO, adding `@config ivar` doesn't add any value, since `SERVICE_CONFIGURATION` constant is already there for accessing service configurations in tests, I would rather use this constant directly inside regex
` assert_match /#{SERVICE_CONFIGURATIONS[:s3][:bucket]}\.s3.(\S+)?amazonaws\.com/, details["url"]`